packages/mural-account-chooser/src/components/account-chooser/index.tsx

```diff
@@ -56,253 +179,205 @@ export enum ACCOUNT_CHOOSER_ACTION {
+    props;
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [page, setPage] = useState<PageName>('Sign in');
+  const [account, setAccount] = useState<Account>();
 
-        if (accountExist) {
-          hintEmailSignIn = { email: hint, authMode };
-        } else {
-          hintEmailSignUp = {
+  useEffect(() => {
+    (async () => {
+      // Go directly to the murally sign up page
+      if (!hint && !visitor) await createNewAccount();
+
+      if (hint) {
+        const realm = await loadRealm(hint);
+        if (realm) {
+          const authMode = getAuthMode(realm);
+          const accountExist =
+            realm.accountStatus === AccountStatus.VALID ||
+            realm.accountStatus === AccountStatus.UNVERIFIED;
+          setAccount({
             email: hint,
             authMode,
-            requireConsent: realm?.requireConsent,
-          };
+            requireConsent: accountExist ? undefined : realm?.requireConsent,
+          });
         }
       }
-    }
-
-    if (silent) return this.autoChoose(hintEmailSignIn, hintEmailSignUp);
 
-    this.setState({
-      hintEmailSignIn,
-      hintEmailSignUp,
-      isLoading: false,
-    });
-  }
+      setIsLoading(false);
+    })();
+  }, []);
 
-  loadRealm = async (email: string) => {
-    const { apiClient } = this.props;
+  const loadRealm = async (email: string) => {
     try {
       return await getMuralRealm(apiClient, email);
     } catch (e: any) {
-      this.props.onError(e);
+      onError(e);
       return null;
     }
   };
 
-  autoChoose = (
-    hintEmailSignIn: StateTypes['hintEmailSignIn'],
-    hintEmailSignUp: StateTypes['hintEmailSignUp'],
-  ) => {
-    const { activeSession } = this.props;
-    if (activeSession) return activeSession.onSelect();
-    if (hintEmailSignIn) return this.hintEmailSignIn();
-    if (hintEmailSignUp) {
-      if (hintEmailSignUp.requireConsent) return this.hintSsoSignUp();
-      return this.hintEmailSignUp();
-    }
-    return this.useAnotherAccount();
+  const createNewAccount = async () => {
+    onSelection(
+      await getAuthUrl({ signup: true }),
+      ACCOUNT_CHOOSER_ACTION.NEW_ACCOUNT,
+    );
   };
 
-  onSelection = (url: string, action?: ACCOUNT_CHOOSER_ACTION) => {
-    this.props.onSelection(url, action);
-  };
+  const useAnotherAccount = async () =>
+    onSelection(await getAuthUrl(), ACCOUNT_CHOOSER_ACTION.ANOTHER_ACCOUNT);
 
-  hintEmailSignIn = async () =>
-    this.onSelection(
-      await this.props.getAuthUrl({
-        auto: {
-          action: 'signin',
-          email: this.props.hint!,
-        },
-      }),
-      ACCOUNT_CHOOSER_ACTION.SIGN_IN,
-    );
+  const continueWithEmail = async () => {
+    if (!hint) {
+      createNewAccount();
+      return;
+    }
+
+    // The hint email is already associated with a mural account
+    // so sign in
+    if (account?.authMode === AuthMode.PASSWORD) {
```

Comment:
This condition is not exactly true. Sometimes, the `authMode` is `Password` but there's no mural account associated with the hint.

Instead, we should look for `accountExist`. So I suggest to add an `accountExist` boolean in the `Account` state.

To validate this use case you can try with an email like `no_account@no_sso_domain.com`. Clicking on `Continue` should open the signup page directly.

Reply:
Oh! How do we tell if the account exists? These are the only fields I am currently getting from the realm response. Is there a `accountExist` field we are missing?

```
export interface RealmResponse {
  accountStatus: AccountStatus;
  canAccessTenant: boolean;
  authUrl: string;
  identityProviderName: string;
  overridable: boolean;
  requireConsent: boolean;
}

export enum AccountStatus {
  UNKNOWN = -1,
  UNKNOWN_SSO,
  UNVERIFIED,
  VALID,
  UNKNOWN_BLOCKED,
}
```

Reply:
Oh I see `accountExist` in the useEffect